### PR TITLE
Avoid class state by passing relevant config to PodReflector on instanciation

### PR DESCRIPTION
**Pull request by dolfinus**
Based on this discussion:
https://github.com/jupyterhub/kubespawner/pull/654#discussion_r1019049868

### CODE REVIEW

**Comment by consideRatio** [comment]
Thanks @dolfinus!! Retriggereing tests to run against main branch after merged #654

File: kubespawner/spawner.py [hunk 1]
```diff
@@ -55,11 +55,6 @@ class PodReflector(ResourceReflector):
 
     kind = "pods"
 
-    # The default component label can be over-ridden by specifying the component_label property
-    labels = {
-        'component': 'singleuser-server',
-    }
-
```
Comment (consideRatio) [inline kubespawner/spawner.py:62]: This is an override of the base class ResourceReflector definition of labels, which is now being passed via `_start_reflector(labels=...)` instead. Presumably, the PodReflector or other base class then should have a constructor accepting that keyword argument...

I concluded that the class inheritance is like this: PodReflector -> ResourceReflector -> LoggingConfigurable -> Configurable -> HasTraits, and that HasTraits does indeed consume it. Okay, this makes sense then!

This is the kind of investigation I felt was needed, making this change something I appreciated having in a standalone PR.

**Review by consideRatio — APPROVED** [review]

File: kubespawner/spawner.py (hunk 1 above)
Comment (dolfinus) [reply kubespawner/spawner.py:62]: I used this as a reference:
https://github.com/jupyterhub/kubespawner/blob/c6bb2a1e3a0ddec6044b4ef8cb31f8fa50571c7c/kubespawner/proxy.py#L271-L279

Comment (consideRatio) [reply kubespawner/spawner.py:62]: I appreciated that! It made us consistent, something I also checked for when reviewing this.

**Pull request merged**